### DISCUSSION TITLE
Question: Split up variants

### DISCUSSION THREAD

**Discussion by sarmad1995** [Shopify/restyle#205] [Ideas]
I just wanted to know if there is any future scope for split variants, currently the variants sit in a single place and can bloat the theme.
Also while developing it doesn't make sense to switch to theme and component always.
Having close variants can also help in some design patterns.
Example

TextInput.ts
TextInput.variants.ts

Where the variants will be somehow imported in the main `Theme`.

**Comment by LouisIV** [comment]
Having a best practice way of doing this would be really great!
